File: Domain/Entities/Locations/Venue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Mappen.Domain.Entities.Locations
{
	public enum SubaddressType
	{ 
		Room,
		Apartment,
		Floor,
		Unit,
		Suite
	}

	public class Venue
	{
		public int Id { get; set; }
		public Address Address { get; set; }
		public string Name { get; set; }
		public long Phone { get; set; }
		public string Email { get; set; }
		public SubaddressType SubaddressType { get; set; }
		public int SubaddressNumber { get; set; }

		public Venue Sync(Venue newVenue)
		{
			Address.Sync(newVenue.Address);
			Name = newVenue.Name;
			Phone = newVenue.Phone;
			Email = newVenue.Email;
			SubaddressType = newVenue.SubaddressType;
			SubaddressNumber = newVenue.SubaddressNumber;
			return this;
		}
	}
}

File: Application/Mappen/Scripts/App/Venue.js
var Venue;
(function () {
    var instances = {};
    Venue = function (id) {
        if (id && instances[id])
            return instances[id];

        this.Id = ko.observable();
        this.Address = new Address();
        this.Name = ko.observable();
        this.Phone = ko.observable();
        this.Email = ko.observable();
        this.SubaddressType = ko.observable();
        this.SubaddressNumber = ko.observable();

        if (id)
            instances[id] = this;
    }

    Venue.prototype = {
        ProcessJson: function (json) {
            this.Address.ProcessJson(json.Address);
            this.Name(json.Name);
            this.Phone(json.Phone);
            this.Email(json.Email);
            this.SubaddressType(json.SubaddressType);
            this.SubaddressNumber(json.SubaddressNumber);
        }
    }
}());